Terminal.Gui/Core/View.cs

```diff
@@ -253,8 +258,8 @@ internal enum Direction {
 		/// <summary>
 		/// Configurable keybindings supported by the control
 		/// </summary>
-		private Dictionary<Key,Command> KeyBindings { get; set; } = new Dictionary<Key, Command>();
-		private Dictionary<Command,Action> CommandImplementations { get; set; } = new Dictionary<Command, Action> ();
+		private Dictionary<Key, Command> KeyBindings { get; set; } = new Dictionary<Key, Command> ();
+		private Dictionary<Command, Func<object [], bool>> CommandImplementations { get; set; } = new Dictionary<Command, Func<object [], bool>> ();
```

Comment:
I'm a bit reluctant to do change this to arbitrary length parameters.  I will merge this but update it to see if I can find a way to keep it as a `Action` if thats ok?

Reply:
I think we want the Command to be `Activate` or something.  So that it is general enough to apply to multiple controls (you can activate a button, a menu bar item etc).  

I think we don't want a command for 'a hotkey was pressed and it was this key' because there is already a system for that in place and theres already a way to change the hotkey on buttons etc

Reply:
Ok, but since I started using Unit Tests to implement the Key Bindings I came to the conclusion that not always an executed command returns true, depending on some conditions, distorting its functionality. So far I have not even started in the most complex views. If we have to compare with the keyboard events, then Key Bindings will not make much sense, I think.

Reply:
> I think we want the Command to be `Activate` or something. So that it is general enough to apply to multiple controls (you can activate a button, a menu bar item etc).

Do you mean making it general to all views? In this case it must be implemented with virtual methods in the `View` class and only be consumed by derived class that override them?

Reply:
> The goal of this feature is to support changing keybindings. For example `Enter` key in a button activates the button. So we should support changing this with the keybinding system. But I don't think we need to move HotKey / ColdKey into the keybinding system especially if it complicates the simplicity of the `Action` type on `CommandImplementations`.

So, you mean not using them in `ProcessHotKey`/`ProcessColdKey` events and only using them on `ProcessKey`? I confess that I was a bit confusing but I think I will be able to understand your point of view :-)

Reply:
I've changed it to `Func<KeyEvent,bool>` for now (fd9e02afe1d446cf86a8f23419b1fb06908c8d5a) since that seems to work for all our use cases so far.   But I am still hopeful for just `Func<bool>` but will need to dig a bit more to see why that's not currently possible.